File: Python/Logica/principal.py
import sys
import os

# Agregar el directorio "Logica" al path para evitar errores de importación
sys.path.append(os.path.abspath(os.path.dirname(__file__) + "/Logica"))

# Importación de módulos corregidos
from Horarios import Horarios
from Paradas import Rutas
from optRutas import optRutas
from BuscarParada import BuscarParadas

class Principal:
    def main(self):
        llamada_horarios = Horarios()
        llamada_pa_ru = Rutas()
        llama_op_rutas = optRutas()
        buscador = BuscarParadas()

        while True:
            print("\n=== BUSES UTPL ===")
            print("1. Horarios")
            print("2. Rutas y Paradas")
            print("3. Rutas Optimizadas")
            print("4. Buscar por Parada")
            print("5. Salir")
            opcion = input("Escoja una opcion: ")

            try:
                opcion = int(opcion)
                if opcion == 1:
                    print("\n=== HORARIOS ===")
                    llamada_horarios.mostrar_horarios()
                elif opcion == 2:
                    print("\n=== RUTAS Y PARADAS ===")
                    llamada_pa_ru.mostrar_resultados()
                elif opcion == 3:
                    print("\n=== RUTAS OPTIMIZADAS ===")
                    llama_op_rutas.mostrar_resultados()
                elif opcion == 4:
                    nombre_parada = input("\nIngrese el nombre de la parada: ").strip()
                    print("\n=== BUSQUEDA DE PARADA ===")
                    buscador.buscar_por_parada(nombre_parada)
                elif opcion == 5:
                    print("\nSaliendo del programa...")
                    break
                else:
                    print("\n Opción no válida. Intente de nuevo.")
            except ValueError:
                print("\n Error: Entrada no válida. Intente de nuevo.")

if __name__ == "__main__":
    principal = Principal()
    principal.main()

File: Sistema_Buses_Proyecto/src/Logica/Principal.java
package Logica;

import java.util.Scanner;

public class Principal {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int opc;

        HorariosRutas llamadaHorarios = new HorariosRutas();
        ParadasRutas llamadaPaRu = new ParadasRutas();
        Opt_Rutas llamaOpRutas = new Opt_Rutas();
        BuscarParadas buscador = new BuscarParadas();

        while (true) {
            System.out.println("=== BUSES UTPL ===");
            System.out.println("1. Horarios");
            System.out.println("2. Rutas y Paradas");
            System.out.println("3. Rutas Optimizadas");
            System.out.println("4. Buscar por parada");
            System.out.println("5. Salir");
            System.out.print("Escoja una opcion: ");

            try {
                opc = input.nextInt();
                input.nextLine(); // Limpiar el buffer del scanner

                switch (opc) {
                    case 1:
                        llamadaHorarios.mostrarHorarios();
                        break;
                    case 2:
                        llamadaPaRu.mostrarResultados();
                        break;
                    case 3:
                        llamaOpRutas.mostrarResultados();
                        break;
                    case 4:
                        System.out.print("Ingrese el nombre de la parada: ");
                        String nombreParada = input.nextLine();
                        buscador.buscarPorParada(nombreParada);
                        break;
                    case 5:
                        System.out.println("Saliendo del programa...");
                        input.close();
                        return;
                    default:
                        System.out.println("Opción no válida. Intente de nuevo.");
                }
            } catch (Exception e) {
                System.out.println("Error: Entrada no válida. Intente de nuevo.");
                input.nextLine(); // Limpiar el buffer del scanner en caso de error
            }
        }
    }
}